packages/app-account/src/containers/InteractiveChains.js

```diff
@@ -4,27 +4,32 @@ import { Grid, Row } from 'react-bootstrap'
 import { connect } from 'react-redux'
 import { reset } from 'redux-form'
 import queryString from 'query-string'
-import { chainDispatch } from 'interbit-ui-tools'
+import { interbitRedux } from 'interbit-ui-tools'
+import { LinkedCovenant } from 'interbit-ui-components'
 
-import LinkedCovenant from '../components/LinkedCovenant'
+import { PRIVATE } from '../constants/chainAliases'
 import { actionCreators } from '../adapters/my-account.adapter'
-import { getExploreChainState } from '../redux/exploreChainReducer'
+
+const { chainDispatch, selectors } = interbitRedux
 
 const mapStateToProps = (state, ownProps) => {
   const {
     location: { search }
   } = ownProps
   const query = queryString.parse(search)
-  const { alias: chainAlias } = query
+  const { alias } = query
 
-  const exploreChainState = getExploreChainState(state, chainAlias)
+  const chainAlias = alias || PRIVATE
+  const chainState = selectors.getChain(state, { chainAlias })
+  const chainId = selectors.getChainId(state, { chainAlias })
 
   return {
     selectedChain: {
-      chainAlias: exploreChainState.chainId,
+      chainId,
+      chainAlias,
       state: {
-        ...exploreChainState.state,
-        interbit: exploreChainState.interbit
+        ...chainState,
+        interbit: chainState.interbit
```

Comment:
Well it could, but... You missed the point here. The spreading is intentional as it brings the application state (which you probably care about the most) to the top and pushes the interbit state to the bottom.